streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -124,6 +124,10 @@ public void openDB(ProcessorContext context) {
         options.setCreateIfMissing(true);
         options.setErrorIfExists(false);
         options.setInfoLogLevel(InfoLogLevel.ERROR_LEVEL);
+        // this is the recommended way to increase parallelism in RocksDb
+        // note that the current implementation increases the number of compaction threads
+        // but not flush threads.
+        options.setIncreaseParallelism(Runtime.getRuntime().availableProcessors());
```

Comment:
@xvrl Yes, there will be more RocksDb threads, but hopefully they won't all be busy at the same time compacting. The alternative is to settle for sub-optimal performance for each RocksDb store. If many threads turns out to be a problem in practice we can revisit. Thanks.